### PR TITLE
User Profile

### PR DESCRIPTION
**Pull request by dndquynh**
- Added edit profile button
- Created an overlay that displays the edit profile form
- Users can enter their name, city, country, descriptions about themselves and their recommendation about their city.
- Updated Datastore and AboutMeServlet to handle new attributes in User class.

Screen Shot 2019-06-28 at 4 04 53 PM
Screen Shot 2019-06-28 at 4 43 55 PM

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -117,13 +117,18 @@ public void storeMessage(Message message) {
...
   public void storeUser(User user) {
     Entity userEntity = new Entity("User", user.getEmail() );
     userEntity.setProperty("email", user.getEmail() );
+    userEntity.setProperty("firstName", user.getFirstName() );
+    userEntity.setProperty("lastName", user.getLastName() );
+    userEntity.setProperty("country", user.getCountry() );
+    userEntity.setProperty("city", user.getCity() );
     userEntity.setProperty("aboutMe", user.getAboutMe() );
-    
+    userEntity.setProperty("recommendation", user.getRecommendation() );
```
Comment (memo-saldana) [inline src/main/java/com/google/codeu/data/Datastore.java:130]: Is this the review the user writes? If so, shouldn't it be separate so a user can have multiple reviews?

File: src/main/webapp/user-page.html [hunk 2]
```diff
@@ -16,38 +16,79 @@
...
+
+            <div class="form-group">
+              <textarea name="about-me" placeholder="About me" class="form-control" rows=3 required></textarea>
+            </div>
+
+            <div class="form-group">
+              <textarea class="form-control" name="recommendation" placeholder="Best place to visit in your city?" rows=3 required></textarea>
+            </div>
+
+            <input type="submit" value="Submit" class="rounded" style="float:right; margin-right:20px; display: inline-block">
```
Comment (memo-saldana) [inline src/main/webapp/user-page.html]: Nit: The style should be made a CSS class

File: src/main/webapp/js/user-page-loader.js [hunk 3]
```diff
@@ -86,7 +84,13 @@ function buildMessageDiv(message) {
   const bodyDiv = document.createElement('div');
   bodyDiv.classList.add('message-body');
   bodyDiv.classList.add('padded');
-	bodyDiv.innerHTML = message.text + message.imageUrl;
+  // If user doesn't attach image file, dont' include the imageUrl in the message body.
+  if (message.imageUrl == '<img src="null" />') {
```
Comment (memo-saldana) [inline src/main/webapp/js/user-page-loader.js:88]: This is great, I should actually add this to the landmark images code

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (dndquynh) [reply src/main/java/com/google/codeu/data/Datastore.java:130]: No, the reviews for Paris will be written in the message box below. This is just their recommendation about where to go in their area, which I put in the About me section.

File: src/main/webapp/user-page.html (hunk 2 above)
Comment (dndquynh) [reply src/main/webapp/user-page.html]: Done!

**Review by edreichua — APPROVED** [review]
LGTM

**Pull request merged**